mypy/fixup.py

```diff
@@ -286,6 +286,11 @@ def lookup_qualified_stnode(modules: Dict[str, MypyFile], name: str,
         if not rest:
             return stnode
         node = stnode.node
+        # In fine-grained mode, could be a cross-reference to a deleted module
+        if node is None:
+            if not quick_and_dirty:
+                assert rest, "Cannot find %s" % (name,)
```

Comment:
I am not sure I understand why it is `assert rest` here. Shouldn't it be `assert node`? Or is this assert necessary at all?

Also another random idea, since `stale_info` was initially designed only for `--quick` mode, but now is used also in fine grained, maybe it makes sense to check that it never gets to be printed in an error message if not in `--quick` mode by having an assert in `TypeInfo.__str__` (or wherever it is converted to string for an error)?

Reply:
The assertion seems copied from above. Since we return if `not rest` is true on line 286, this assert will always be true and can perhaps be dropped? `assert node` would always fail since we just checked for `node is None`.

Reply:
Oh, oops. This is supposed to be `assert node`, which will always fail. That seems to be the existing idiom in fixup: do a check, assert the check if it is not quick_and_dirty, return a failure if it is quick_and_dirty.

Reply:
Some sort of check to prevent it seeping into error messages would be good, but it's not super obvious what the best way would be. Putting an assert in `__str__` I don't think would work, since it *might* get converted to an error string, but the errors should always get replaced with something else while reprocessing